arangod/Aql/EngineInfoContainerDBServerServerBased.cpp

```diff
@@ -447,48 +456,53 @@ Result EngineInfoContainerDBServerServerBased::parseResponse(
  * they may be leftovers from Coordinator.
  * Will also clear the list of queryIds after return.
  *
- * @param cc The ClusterComm
+ * @param pool The ConnectionPool
  * @param errorCode error Code to be send to DBServers for logging.
  * @param dbname Name of the database this query is executed in.
  * @param queryIds A map of QueryIds of the format: (remoteNodeId:shardId)
  * -> queryid.
  */
 void EngineInfoContainerDBServerServerBased::cleanupEngines(
-    std::shared_ptr<ClusterComm> cc, int errorCode, std::string const& dbname,
+    network::ConnectionPool* pool, int errorCode, std::string const& dbname,
     MapRemoteToSnippet& queryIds) const {
+  network::RequestOptions options;
+  options.timeout = network::Timeout(10.0);  // Picked arbitrarily
+  network::Headers headers;
+
   // Shutdown query snippets
   std::string url("/_db/" + arangodb::basics::StringUtils::urlEncode(dbname) +
                   "/_api/aql/shutdown/");
-  std::vector<ClusterCommRequest> requests;
-  auto body = std::make_shared<std::string>(
-      "{\"code\":" + std::to_string(errorCode) + "}");
+  VPackBuffer<uint8_t> body;
+  VPackBuilder builder(body);
+  builder.openObject();
+  builder.add("code", VPackValue(std::to_string(errorCode)));
+  builder.close();
   for (auto const& it : queryIds) {
     // it.first == RemoteNodeId, we don't need this
     // it.second server -> [snippets]
     for (auto const& serToSnippets : it.second) {
       auto server = serToSnippets.first;
       for (auto const& shardId : serToSnippets.second) {
-        requests.emplace_back(server, rest::RequestType::PUT, url + shardId, body);
+        // fire and forget
+        network::sendRequest(pool, server, fuerte::RestVerb::Put, url + shardId,
+                             body, headers, options);
       }
+      _query.incHttpRequests(serToSnippets.second.size());
     }
   }
 
   // Shutdown traverser engines
   url = "/_db/" + arangodb::basics::StringUtils::urlEncode(dbname) +
         "/_internal/traverser/";
-  std::unordered_map<std::string, std::string> headers;
-  std::shared_ptr<std::string> noBody;
-
-  CoordTransactionID coordinatorTransactionID = TRI_NewTickServer();
-  auto cb = std::make_shared<::NoopCb>();
+  VPackBuffer<uint8_t> noBody;
 
-  constexpr double shortTimeout = 10.0;  // Picked arbitrarily
   for (auto const& gn : _graphNodes) {
     auto allEngines = gn->engines();
     for (auto const& engine : *allEngines) {
-      cc->asyncRequest(coordinatorTransactionID, engine.first, rest::RequestType::DELETE_REQ,
-                       url + basics::StringUtils::itoa(engine.second), noBody,
-                       headers, cb, shortTimeout, false, 2.0);
+      // fire and forget
+      network::sendRequestRetry(pool, engine.first, fuerte::RestVerb::Delete,
+                                url + basics::StringUtils::itoa(engine.second),
+                                noBody, headers, options);
```

Comment:
So as far as I can tell, it looks like the previous version used an initial timeout of 2 seconds, retrying for up to 10 seconds. We don't have an exact analog for initial timeout with our sendRequestRetry method, but in this case I've chosen to honor the total timeout rather than the initial.

Reply:
Alright then.